sim/rogue/items_sets_pve.go

```diff
@@ -213,3 +215,175 @@ var ItemSetNightSlayerBattlearmor = core.NewItemSet(core.ItemSet{
+				if spell.ProcMask.Matches(core.ProcMaskMeleeMH) {
+					spell.DamageMultiplier *= 1.10
+				}
+			})
+		},
+		// Reduces cooldown on vanish by 2.5 minutes
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.RegisterAura(core.Aura{
+ 				Label: "S03 - Item - T2 - Rogue - Damage 6P Bonus",
+  				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+    				rogue.Vanish.CD.Duration -= time.Second * 150
+  				},
+			})
+		},
+	},
+})
+
+var ItemSetBloodfangBattlearmor = core.NewItemSet(core.ItemSet{
+	Name: "Bloodfang Battlearmor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Rolling with the Punches now also activates every time you gain a combo point.
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+				return
+			}
+
+			rogue.OnComboPointsGained(func(sim *core.Simulation) {
+				if rogue.RollingWithThePunchesProcAura.GetStacks() == 5 {
+					rogue.RollingWithThePunchesProcAura.Refresh(sim)
+				} else if rogue.RollingWithThePunchesProcAura.IsActive() {
+					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+				} else {
+					rogue.RollingWithThePunchesProcAura.Activate(sim)
+					rogue.RollingWithThePunchesProcAura.AddStack(sim)
+				}
+			})
+		},
+		// Your Rolling with the Punches also grants you 20% increased Armor from items per stack.
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+				return
+			}
+			initarmor := rogue.BaseEquipStats()[stats.Armor]
+
+			rogue.RegisterAura(core.Aura{
+				Label: "T2 Please Replace",
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+				  oldOnStacksChange := rogue.RollingWithThePunchesProcAura.OnStacksChange
+				  rogue.RollingWithThePunchesProcAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+					oldOnStacksChange(aura, sim, oldStacks, newStacks)
+					rogue.AddStatDynamic(sim, stats.Armor, float64(0.2*initarmor*float64(newStacks-oldStacks)))
+				}
+				},
+			  })
+		},
+		// The cooldown on your Main Gauche resets every time your target Dodges or Parries.
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneMainGauche) {
+				return
+			}
+
+			core.MakePermanent(rogue.RegisterAura(core.Aura{
+				Label:    "S03 - Item - T2 - Rogue - Tank 6P Bonus",
+				OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if result.DidDodge() || result.DidParry() {
+						rogue.MainGauche.CD.Reset()
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetMadCapsOutfit = core.NewItemSet(core.ItemSet{
+	Name: "Madcap's Outfit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// +20 Attack Power
+		2: func(agent core.Agent) {
+			c := agent.GetCharacter()
+			c.AddStats(stats.Stats{
+				stats.AttackPower:       20,
```

Comment:
This set bonus applies both AttackPower and RangedAttackPower according to wowhead